### ISSUE TITLE
load_resource_pack only works a limited number of times

### ISSUE THREAD

**Issue by iactix** [godotengine/godot#28642]
**Godot version:**
v3.1.stable.official

**OS:**
Kubuntu 18.10

**Issue description:**
ProjectSettings.load_resource_pack stops working after calling it ~1000 times (in my case). It doesn't have to be different packs, just the same one over and over again will do. It seems to leave every pack file open, including multiple copies of the same pack. It therefore fails when the limit of open files per process is reached. I increased that limit on my machine and then it took more calls to fail.

That means every application that uses this call in a way that can be triggered by the user will stop working sooner or later.

**Steps to reproduce:**
Call ProjectSettings.load_resource_pack() on any pack over and over again

**Comment by akien-mga** [comment]
This might be enough to fix it (untested):
```diff
diff --git a/core/io/file_access_pack.cpp b/core/io/file_access_pack.cpp
index d38d09c6b..930162741 100644
--- a/core/io/file_access_pack.cpp
+++ b/core/io/file_access_pack.cpp
@@ -201,6 +201,7 @@ bool PackedSourcePCK::try_open_pack(const String &p_path) {
 		PackedData::get_singleton()->add_path(p_path, path, ofs, size, md5, this);
 	};
 
+	memdelete(f);
 	return true;
 };
```

**Comment by akien-mga** [comment]
Fixed by 2bc7f9e545ba9f8cb3e77d2607a2cfe33a716bbd.
